contentgrid-spring-boot-actuators/src/main/java/com/contentgrid/spring/boot/actuator/webhooks/WebhookVariables.java

```diff
@@ -0,0 +1,29 @@
+package com.contentgrid.spring.boot.actuator.webhooks;
+
+import com.contentgrid.spring.boot.actuator.ContentGridApplicationProperties.SystemProperties;
+import java.util.Map;
+import lombok.Builder;
+import lombok.Value;
+import org.springframework.util.PropertyPlaceholderHelper.PlaceholderResolver;
+
+@Value
+@Builder
+public class WebhookVariables implements PlaceholderResolver {
+    SystemProperties systemProperties;
+    Map<String, String> userVariables;
+
+    @Override
+    public String resolvePlaceholder(String placeholderName) {
+        if(placeholderName.startsWith("vars.")) {
+            return userVariables.get(placeholderName.substring("vars.".length()));
+        }
+        switch(placeholderName) {
+            case "system.application.id":
+                return systemProperties.getApplicationId();
+            case "system.deployment.id":
+                return systemProperties.getDeploymentId();
+            default:
+                throw new IllegalArgumentException(String.format("Can not find a replacement for placeholder '%s'", placeholderName));
```

Comment:
that kind of violates the `PlaceholderResolver` contract. I'm not sure this is the correct place to throw an error when some placeholder cannot be resolved.